### PR TITLE
Add support for Maven 3.5 and 3.6

### PR DESCRIPTION
**Pull request by jkutner**

### CODE REVIEW

File: lib/common.sh [hunk 1]
```diff
@@ -30,21 +30,16 @@ download_maven() {
   local installDir=$2
   local mavenHome=$3
   rm -rf $mavenHome
-  curl --retry 3 --silent --max-time 60 --location ${mavenUrl} | tar xzm -C $installDir
+  curl --retry 3 --silent --max-time 60 --location "${mavenUrl}" | tar xzm -C $installDir
```
Comment (schneems) [inline lib/common.sh]: This curl is subject to error 18 which won't be retryable (I think when the connection cuts out in the middle of a download) PHP buildpack uses a custom function for curling binaries `curl_retry_on_18`. Might be useful in the future

File: vendor/maven/sources.txt [hunk 2]
```diff
@@ -5,3 +5,6 @@ https://apache.org/dist/maven/maven-3/3.2.5/binaries/apache-maven-3.2.5-bin.tar.
 https://apache.org/dist/maven/maven-3/3.3.1/binaries/apache-maven-3.3.1-bin.tar.gz
 https://apache.org/dist/maven/maven-3/3.3.3/binaries/apache-maven-3.3.3-bin.tar.gz
 https://apache.org/dist/maven/maven-3/3.3.9/binaries/apache-maven-3.3.9-bin.tar.gz
+https://apache.org/dist/maven/maven-3/3.3.9/binaries/apache-maven-3.3.9-bin.tar.gz
```
Comment (schneems) [inline vendor/maven/sources.txt]: This is in here twice, not sure if that's a problem or not.

Comment (jkutner) [reply vendor/maven/sources.txt]: good catch

File: lib/common.sh (hunk 1 above)
Comment (jkutner) [reply lib/common.sh]: Oh, TIL. I will open a separate issue for this.

**Pull request merged**